sklego/meta.py

```diff
@@ -114,62 +228,129 @@ def fit(self, X, y):
         pred_col = 'the-column-that-i-want-to-predict-but-dont-have-the-name-for'
         if isinstance(X, np.ndarray):
             X = pd.DataFrame(X, columns=[str(_) for _ in range(X.shape[1])])
-        X = X.assign(**{pred_col: y})
+
+        if isinstance(y, np.ndarray):
+            cols = pred_col if y.ndim == 1 else ["_".join([pred_col, i]) for i in range(y.shape[1])]
+            y = pd.Series(y, name=cols) if y.ndim == 1 else pd.DataFrame(y, columns=cols)
 
         self.group_colnames_ = [str(_) for _ in as_list(self.groups)]
+
+        # List of all hierarchical subsets of columns
+        self.group_colnames_hierarchical_ = self.__expanding_list(self.group_colnames_, list)
+
         if any([c not in X.columns for c in self.group_colnames_]):
             raise KeyError(f"{self.group_colnames_} not in {X.columns}")
 
-        self.X_colnames_ = [_ for _ in X.columns if _ not in self.group_colnames_ and _ is not pred_col]
+        self.value_colnames_ = [_ for _ in X.columns if _ not in self.group_colnames_]
         self.fallback_ = None
 
         if self.use_fallback:
-            subset_x = X[self.X_colnames_]
+            subset_x = X[self.value_colnames_]
             self.fallback_ = clone(self.estimator).fit(subset_x, y)
 
-        self.groups_ = X[self.group_colnames_].drop_duplicates()
+        if self.shrinkage:
+            self.estimators_ = {}
+
+            for level_colnames in self.group_colnames_hierarchical_:
+                self.estimators_.update(
+                    self.__fit_grouped_estimator(X, y, self.value_colnames_, level_colnames)
+                )
+        else:
+            self.estimators_ = self.__fit_grouped_estimator(X, y, self.value_colnames_, self.group_colnames_)
+
+        self.groups_ = as_list(self.estimators_.keys())
+
+        if self.shrinkage:
+            self.complete_groups_ = [grp for grp in self.groups_ if len(as_list(grp)) == len(self.group_colnames_)]
```

Comment:
maybe pass `complete_groups_` to `complete_groups_ ` instead of creating an attribute?

Reply:
now I had to do a search to find out where this attribute is created